src/amm/spfFee/contracts/n2tPoolContracts.ts

```diff
@@ -0,0 +1,96 @@
+import {ErgoTree, PublicKey, TokenId} from "@ergolabs/ergo-sdk"
+import {SpecExFeePerToken} from "../../../types"
+import {Bool, Bytes, Int, Long, ProveDlog, RedeemerBytes} from "../../../utils/contract"
+import {decimalToFractional} from "../../../utils/math"
+import {PoolId} from "../../common/types"
+import {DepositContract, RedeemContract, SwapBuyContract, SwapSellContract} from "./n2tTemplates"
+
+export function deposit(
+  poolId: PoolId,
+  pk: PublicKey,
+  selfX: bigint,
+  selfY: bigint,
+  maxMinerFee: bigint
+): ErgoTree {
+  return DepositContract
+    .build({
+      refundProp:        ProveDlog(pk),
+      selfXAmount:       Long(selfX),
+      selfYAmount:       Long(selfY),
+      poolNFT:           Bytes(poolId),
+      redeemerPropBytes: RedeemerBytes(pk),
+      maxMinerFee:       Long(maxMinerFee)
```

Comment:
I like the idea but how to make formatter keep this beautiful indentation?

Reply:
we need to write additional prettifier rules for this. I'll try to solve this issue in separate task